### PR TITLE
Change docker guide link to point to worker tutorial

### PR DESCRIPTION
**Pull request by kevingrismore**
Change docker guide link to point to worker tutorial since the docker guide no longer shows docker-based storage.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: docs/guides/deployment/storage-guide.md [hunk 1]
```diff
@@ -198,7 +198,7 @@ Another popular way to store your flow code is to include it in a Docker image.
     - Azure Container Instances - Push
     - Google Cloud Run - Push
 
-1. Run `prefect init` in the root of your repository and choose `docker` for the project name and answer the prompts to create a `prefect.yaml` file with a build step that will create a Docker image with the flow code built in. See the [Docker guide](/guides/deployment/docker/) for more info.
+1. Run `prefect init` in the root of your repository and choose `docker` for the project name and answer the prompts to create a `prefect.yaml` file with a build step that will create a Docker image with the flow code built in. See the [Docker guide](/tutorial/worker/) for more info.
```
Comment (discdiver) [inline docs/guides/deployment/storage-guide.md]:
```suggestion
1. Run `prefect init` in the root of your repository and choose `docker` for the project name and answer the prompts to create a `prefect.yaml` file with a build step that will create a Docker image with the flow code built in. See the [Workers and Work Pools page of the tutorial](/tutorial/worker/) for more info.
```

Comment (discdiver) [reply docs/guides/deployment/storage-guide.md]: Looks like it's a link to the tutorial page instead of the guide, so updated the text.

**Review by discdiver — CHANGES_REQUESTED** [review]
Thank you! One change to the anchor text.

File: docs/guides/deployment/storage-guide.md (hunk 1 above)
Comment (kevingrismore) [reply docs/guides/deployment/storage-guide.md]: Fixed a typo while you were writing this but it's in the next commit.

**Comment by kevingrismore** [comment]
Some more thoughts on this - the page I'm linking to doesn't have any explicit examples of baking flow code into a Docker image, but both the Docker and Kubernetes pages do. The Docker guide covers the `flow.serve()` deployment method which isn't in theme with where we're linking out from, but the Kubernetes guide doesn't cover Docker images as code storage in a concise way. Might come back around and change this link in the near future.

**Review by discdiver — APPROVED** [review]
lgtm

**Comment by discdiver** [comment]
> Some more thoughts on this - the page I'm linking to doesn't have any explicit examples of baking flow code into a Docker image, but both the Docker and Kubernetes pages do. The Docker guide covers the `flow.serve()` deployment method which isn't in theme with where we're linking out from, but the Kubernetes guide doesn't cover Docker images as code storage in a concise way. Might come back around and change this link or drop an example directly into the storage page in the near future.

Cool.

fwiw - making the options clearer for container-based infrastructure and related storage and deployment guidance is something we want to tackle.

**Review by cicdw — APPROVED** [review]

**Pull request merged**